src/api/transactions.coffee

```diff
@@ -220,12 +228,71 @@ exports.updateTransaction = (transactionId) ->
 
   transactionId = unescape transactionId
   updates = this.request.body
+  that = this
 
   try
-    yield transactions.Transaction.findByIdAndUpdate(transactionId, updates).exec()
-    this.body = "Transaction with ID: #{transactionId} successfully updated"
-    this.status = 'ok'
-    logger.info "User #{this.authenticated.email} updated transaction with id #{transactionId}"
+    yield transactions.Transaction.findByIdAndUpdate transactionId, updates, () ->
```

Comment:
I am seeing some really weird behavior here with the yield and the callback. Errors updating the transaction aren't being handled and I am getting a combination of successful and failed responses and log messages.